examples/convert_ieeg_to_bids.py

```diff
@@ -0,0 +1,151 @@
+# The electrode data are in the Matlab format: '.mat'. This is easy to read in with
+# `scipy.io.loadmat` function. We also need to get some sample EEG data, so we will
+# just generate random data from white noise. Here is where you would use your own data
+# if you had it.
+eegdata = np.random.rand(len(ch_names), 1000)
+
+# However, apart from the data format, we need to build
+# a directory structure and supply meta data
+# files to properly *bidsify* this data.
+info = mne.create_info(ch_names, 1000., 'ecog', montage=montage)
+raw = mne.io.RawArray(eegdata, info)
+
+###############################################################################
+# Step 2: Formatting as BIDS
+# --------------------------
+#
+# Let's start by formatting a single subject.
+###############################################################################
+# With this step, we have everything to start a new BIDS directory using
+# our data. To do that, we can use :func:`write_raw_bids`
+# Generally, :func:`write_raw_bids` tries to extract as much
+# meta data as possible from the raw data and then formats it in a BIDS
+# compatible way. :func:`write_raw_bids` takes a bunch of inputs, most of
+# which are however optional. The required inputs are:
+#
+# * :code:`raw`
+# * :code:`bids_basename`
+# * :code:`bids_root`
+#
+# ... as you can see in the docstring:
+print(write_raw_bids.__doc__)
+
+###############################################################################
+# Let us initialize some of the necessary data for the subject
+# There is a subject, and specific task for the dataset
+subject_id = '001'  # zero padding to account for >100 subjects in this dataset
+task = 'testresteyes'
+bids_root = os.path.join(home, 'mne_data', 'eegmmidb_bids')
+
+###############################################################################
+# Now we just need to specify a few more EEG details to get something sensible:
+# Brief description of the event markers present in the data. This will become
+# the `trial_type` column in our BIDS `events.tsv`. We know about the event
+# meaning from the documentation on PhysioBank
+
+# # Now convert our data to be in a new BIDS dataset.
+bids_basename = make_bids_basename(subject=subject_id,
+                                   task=task,
+                                   acquisition="ecog")
+
+# need to set the filenames if we are initializing data from array
+with tempfile.TemporaryDirectory() as tmp_root:
+    tmp_fpath = os.path.join(tmp_root, "test_raw.fif")
+    raw.save(tmp_fpath)
+    raw = mne.io.read_raw_fif(tmp_fpath)
+    raw_file = raw
+
+    write_raw_bids(raw_file, bids_basename,
+               bids_root,
+               overwrite=True)
+
+###############################################################################
+# Step 3: Check and compare with standard
+# ---------------------------------------
+# Now we have written our BIDS directory.
+print_dir_tree(bids_root)
```

Comment:
do you think we can now add some lines of code to read in the file using `read_raw_bids` and plot the sensor locations to ensure that the round trip was successful?

Reply:
> do you think we can now add some lines of code to read in the file using `read_raw_bids` and plot the sensor locations to ensure that the round trip was successful?

Added in some plotting based on: https://mne.tools/stable/auto_tutorials/misc/plot_ecog.html#sphx-glr-auto-tutorials-misc-plot-ecog-py

I'm not sure how can we render this in the final documentation though?

Reply:
hummm ... I am not sure it's worth it to set up CircleCI with `mayavi` and virtual frame buffering. @agramfort any opinions?

Reply:
agreed. Let's leave mayavi/vtk et al. out of here.